### DISCUSSION TITLE
Passing "method" argument to singlediode

### DISCUSSION THREAD

**Discussion by sixyphe** [pvlib/pvlib-python#1802] [Q&A] [answered]
Greetings team!  One of my test weather sets, which used to work fine under pvlib 0.8.0 and 0.9.0, has started misbehaving when using more recent versions.
I've tracked down the issue to what appears to be a non-convergence when using **lambertw_i_from_v()**.  If I request the _newton_ or _brentq_ methods when calling **singlediode()**, everything works well.  However, the only way I've found to do this is to actually change the default method used when calling **singlediode()** in pvsystem.py (0.9.5:pvsystem.py:2712).  I've been looking for a more elegant way to select a different method when building the PVSystem, but my exploration of the code has been fruitless so far.
My question is therefore this: when I build my PVSystem (or my Array or ModelChain, but that seems unlikely), is there a way to provide the "method" argument that I want **singlediode()** to use?
Thank you :-)

**Comment by kandersolar** [comment, marked as answer]
Hello @sixyphe,

I think you're right that there is currently no clean way to specify the singlediode method when using the PVSystem class.  That would be a good feature to add.

However that might not be necessary to resolve your problem.  Some changes (see #1782 and the issues linked therein) were made to the lambert solver in the last release that might solve your issue.  Can you install v0.10.1 (`pip install pvlib==0.10.1`) and see if your issue persists?

**Reply by sixyphe** [reply]
Thank you for the quick response!
I installed 0.10.1 yesterday and started testing - there are new issues that pop up with that data set, I'm working on sorting those out (it's a tmy I generate with get_pvgis_tmy in the epw format, but it's a bit wonky, I must admit, dni values are all over the place).  I'll keep you guys posted if I hit a dead end.
I'm fairly new at python, but I have a good deal of general programming experience.  Once I feel more comfortable, I'd be happy to submit some contributions.
